### ISSUE TITLE
Callsite representation (avoid pointing to wrappers.c)

### ISSUE THREAD

**Issue by artpol84** [LLNL/mpiP#9]
For the following application ([ring_c.txt](https://github.com/LLNL/mpiP/files/3263854/ring_c.txt)), I have to set the stack depth:
```
export MPIP="-k 2")
```
To see the actual callsites and get the following output:
```
---------------------------------------------------------------------------
@--- Callsites: 6 ---------------------------------------------------------
---------------------------------------------------------------------------
 ID Lev File/Address        Line Parent_Funct             MPI_Call
  1   0 wrappers.c           515 MPI_Barrier              Barrier
  1   1 ring_c.c              81 main
  2   0 wrappers.c           515 MPI_Barrier              Barrier
  2   1 ring_c.c              78 main
  3   0 wrappers.c          6649 MPI_Recv                 Recv
  3   1 ring_c.c              52 main
  4   0 wrappers.c          7608 MPI_Send                 Send
  4   1 ring_c.c              60 main
  5   0 wrappers.c          6649 MPI_Recv                 Recv
  5   1 ring_c.c              71 main
  6   0 wrappers.c          7608 MPI_Send                 Send
  6   1 ring_c.c              39 main
```
An entry in the wrapper.c as it points to mpiP (profiler tool) internals that are irrelevant to the application being analyzed.
Does it make sense to always shift one stack frame up to skip wrappers.c?

**Comment by artpol84** [comment]
Clarification: this is with libunwind

**Comment by artpol84** [comment]
https://github.com/LLNL/mpiP/pull/10 seems to fix the issue for me.
Though I noticed the following comment:
```
// Inlining / noinlining may affect frames in report
//   - Tools frames may appear in report
```
I'm not sure if this is a concern here: mpiP can control whether or not it's functions are inlined. It is possible to have a NO_INLINE macro that will restrict compiler from inlining functions that are capturing stack.

Anyway, I believe that other variants of stack capturing are using this approach, so I assume it is safe to use it here as well.

**Comment by artpol84** [comment]
#10 turned to be incorrect.
As the replacement, I pushed fdafcc8450be26415beb605cc520cac379ecfd94.
As opposed to indirectly guessing how many frames to skip, this commit is attacking it directly: search for the `MPI_op` in the stacktrace to identify the beginning of what should be reported.

**Comment by artpol84** [comment]
Current version report:
```
---------------------------------------------------------------------------
@--- Callsites: 3 ---------------------------------------------------------
---------------------------------------------------------------------------
  1   0 7-nonblock-waitall.c    48 func1                    Waitall
  1   1 7-nonblock-waitall.c    53 func2
  1   2 7-nonblock-waitall.c    57 func3
  1   3 7-nonblock-waitall.c    64 main
  2   0 wrappers.c            6218 MPI_Isend                Isend
  2   1 7-nonblock-waitall.c    47 func1
  2   2 7-nonblock-waitall.c    53 func2
  2   3 7-nonblock-waitall.c    57 func3
  2   4 7-nonblock-waitall.c    64 main
  3   0 7-nonblock-waitall.c    46 func1                    Irecv
  3   1 7-nonblock-waitall.c    53 func2
  3   2 7-nonblock-waitall.c    57 func3
  3   3 7-nonblock-waitall.c    64 main
```
`wrappers.c` is captured for `Isend` but not for others (build was with `-O3`).

With this patch:
```
---------------------------------------------------------------------------
@--- Callsites: 3 ---------------------------------------------------------
---------------------------------------------------------------------------
 ID Lev File/Address        Line Parent_Funct             MPI_Call
  1   0 7-nonblock-waitall.c    46 func1                    Irecv
  1   1 7-nonblock-waitall.c    53 func2
  1   2 7-nonblock-waitall.c    57 func3
  1   3 7-nonblock-waitall.c    64 main
  2   0 7-nonblock-waitall.c    48 func1                    Waitall
  2   1 7-nonblock-waitall.c    53 func2
  2   2 7-nonblock-waitall.c    57 func3
  2   3 7-nonblock-waitall.c    64 main
  3   0 7-nonblock-waitall.c    47 func1                    Isend
  3   1 7-nonblock-waitall.c    53 func2
  3   2 7-nonblock-waitall.c    57 func3
  3   3 7-nonblock-waitall.c    64 main
```

**Comment by artpol84** [comment]
Addressed by #14 that was merged.
